gneiss/q2/_type.py

```diff
@@ -0,0 +1,23 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, gneiss development team.
+#
+# Distributed under the terms of the GPLv3 License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+from qiime.plugin import SemanticType
+from q2_types.feature_table import (
+    FeatureTable, Frequency, BIOMV100DirFmt, BIOMV210DirFmt)
+
+
+# Regression types
+Regression = SemanticType('Regression', field_names=['type'])
+
+Linear = SemanticType('Linear',
+                      variant_of=Regression.field['type'])
+LinearMixedEffects = SemanticType('LinearMixedEffects',
+                                  variant_of=Regression.field['type'])
+
+# Tree types
+Hierarchy = SemanticType('Hierarchy', field_names=['type'])
+Cluster = SemanticType('Cluster', variant_of=Hierarchy.field['type'])
```

Comment:
In context of the balances, there are quite a few more types.

The tree could be produced to represent a gradient along an environment.
It could also be used to represent the largest variance partitions, analogous to Principle Components Analysis.  We have a few more techniques in the works that will yield even more different ways to generate trees specific for features.

For samples, there is hierarchical clustering, but there is also decision trees.  I'd think it would be a good idea to keep trees relating samples and trees relating features to be distinct entities, but maybe stem from the same super type.